### PR TITLE
Convert spec source into Bikeshed format

### LINKED ISSUE #53: [editorial] Convert the spec source to Bikeshed/ReSpec
As suggested by @xfq in #52 we should probably convert the spec source to Bikeshed/ReSpec to make it easier to publish.

As of now, this spec inherited from Geolocation WG has 29 errors (and 5 warnings) in the Pubrules checker: https://www.w3.org/pubrules/?url=https%3A%2F%2Fw3c.github.io%2Fdeviceorientation%2F&profile=WD-Echidna&validation=simple-validation&noRecTrack=false&informativeOnly=false&echidnaReady=true&patentPolicy=pp2004

### PR DESCRIPTION
**Pull request by anssiko**
Besides style changes, the following changes were made:
- use standard Status of this document for an Editor's Draft
- add missing IDL block to the compassneedscalibration Event section
- update [HTML51] references to [HTML]
- editorial: use Bikeshed standard Feedback header
- editorial: link all occurrences of definitions
- editorial: rename all "Section X" links to "Section title" links
- editorial: split references to normative and informative subsections
- editorial: add IDL index

Fix #53

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/deviceorientation/pull/62.html" title="Last updated on Jan 30, 2019, 2:53 PM UTC (a92514f)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/deviceorientation/62/02426f1...a92514f.html" title="Last updated on Jan 30, 2019, 2:53 PM UTC (a92514f)">Diff</a>

### CODE REVIEW

**Comment by anssiko** [comment]
Note: the pr-preview "Diff" is not very useful due to style changes. The easiest way to review is probably to look at the pr-preview "Preview" version side-by-side with the current spec.

File: index.bs [hunk 1]
```diff
@@ -0,0 +1,654 @@
...
+
+User agents may impose implementation-specific limits on otherwise unconstrained inputs, e.g. to prevent denial of service attacks, to guard against running out of memory, or to work around platform-specific limitations.
+
+Implementations that use ECMAScript to implement the APIs defined in this specification must implement them in a manner consistent with the ECMAScript Bindings defined in the Web IDL specification, as this specification uses that specification's terminology. [[!WEBIDL]]
+
+The events introduced by this specification implement the Event interface defined in the DOM4 Specification, [[!DOM4]]. Implementations must therefore support this specification.
+
+
+
+Introduction {#introduction}
```
Comment (reillyeon) [inline index.bs]: This section is non-normative.

File: index.bs [hunk 2]
```diff
@@ -0,0 +1,654 @@
...
+<div class="example">
+A device is mounted in a vehicle, with the screen in a vertical plane, the top uppermost and facing the rear of the vehicle. The vehicle is travelling at speed v around a right-hand bend of radius r. The device records a positive x component for both {{DeviceMotionEvent/acceleration}} and {{DeviceMotionEvent/accelerationIncludingGravity}}. The device also records a negative value for {{DeviceMotionEvent/rotationRate!!attribute}}.{{DeviceRotationRate/gamma}}:
+<pre class="lang-json">
+{acceleration: {x: v^2/r, y: 0, z: 0},
+ accelerationIncludingGravity: {x: v^2/r, y: 0, z: 9.81},
+ rotationRate: {alpha: 0, beta: 0, gamma: -v/r*180/pi} };
+</pre>
+</div>
+
+Scope {#scope}
```
Comment (reillyeon) [inline index.bs]: This section is non-normative.

File: index.bs [hunk 3]
```diff
@@ -0,0 +1,654 @@
...
+
+The second DOM event provided by this specification, [=devicemotion=], supplies the acceleration of the device, expressed in Cartesian coordinates in a coordinate frame defined in the device. It also supplies the rotation rate of the device about a local coordinate frame. Where practically possible, the event should provide the acceleration of the device's center of mass.
+
+Finally, the specification provides a [=compassneedscalibration=] DOM event, which is used to inform Web sites that a compass being used to provide data for one of the above events is in need of calibration.
+
+The following code extracts illustrate basic use of the events.
+
+<div class="example">
+Registering to receive [=deviceorientation=] events:
+<pre class="lang-json">
```
Comment (reillyeon) [inline index.bs]: s/lang-json/lang-js/

File: index.bs [hunk 4]
```diff
@@ -0,0 +1,654 @@
...
+{alpha: 270 - alpha,
+ beta: 0,
+ gamma: 90};
+</pre>
+</div>
+
+<div class="example">
+Showing custom UI to instruct the user to calibrate the compass:
+<pre class="lang-json">
```
Comment (reillyeon) [inline index.bs]: s/lang-json/lang-js/

File: index.bs [hunk 5]
```diff
@@ -0,0 +1,654 @@
...
+window.addEventListener("compassneedscalibration", function(event) {
+    alert('Your compass needs calibrating! Wave your device in a figure-eight motion');
+    event.preventDefault();
+}, true);
+</pre>
+</div>
+
+<div class="example">
+Registering to receive [=devicemotion=] events:
+<pre class="lang-json">
```
Comment (reillyeon) [inline index.bs]: s/lang-json/lang-js/

File: index.bs [hunk 6]
```diff
@@ -0,0 +1,654 @@
...
+    <img src="equation6.png" alt="theta = atan((v'_x)/(v'_y)) = atan((-cos(alpha)sin(gamma)-sin(alpha)sin(beta)cos(gamma))/(-sin(alpha)sin(gamma)+cos(alpha)sin(beta)cos(gamma)))">
+</object>
+
+provided that &beta; and &gamma; are not both zero.
+
+<div class="example">
+
+The compass heading calculation above can be represented in JavaScript as follows to return the correct compass heading when the provided parameters are defined, not null and represent {{DeviceOrientationEvent/absolute}} values.
+
+<pre class="lang-json">
```
Comment (reillyeon) [inline index.bs]: s/lang-json/lang-js/

File: index.bs [hunk 7]
```diff
@@ -0,0 +1,654 @@
...
+
+If R represents the rotation matrix of the device in the earth frame XYZ, then since the initial body frame is aligned with the earth, R is as follows.
+
+<object class="equation" data="equation13a.xhtml" type="application/mathml+xml">
+    <img src="equation13a.png" alt="R = ZXY = [[cos(alpha) cos(gamma)-sin(alpha) sin(beta) sin(gamma), -cos(beta) sin(alpha), cos(gamma) sin(alpha) sin(beta)+cos(alpha) sin(gamma)], [cos(gamma) sin(alpha)+cos(alpha) sin(beta) sin(gamma), cos(alpha) cos(beta), sin(alpha) sin(gamma)-cos(alpha) cos(gamma) sin(beta)], [-cos(beta) sin(gamma), sin(beta), cos(beta) cos(gamma)]]">
+</object>
+
+<div class="example">
+The above combined rotation matrix can be represented in JavaScript as follows provided passed parameters are defined, not null and represent {{DeviceOrientationEvent/absolute}} values.
+<pre class="lang-json">
```
Comment (reillyeon) [inline index.bs]: s/lang-json/lang-js/

File: index.bs [hunk 8]
```diff
@@ -0,0 +1,654 @@
...
+
+If q represents the unit quaternion of the device in the earth frame XYZ, then since the initial body frame is aligned with the earth, q is as follows.
+
+<object class="equation" data="equation14.xhtml" type="application/mathml+xml">
+    <img src="equation14.png" alt="q = [[q_w], [q_x], [q_y], [q_z]] = [[cos(beta)cos(gamma)cos(alpha) - sin(beta)sin(gamma)sin(alpha)], [sin(beta)cos(gamma)cos(alpha) - cos(beta)sin(gamma)sin(alpha)], [cos(beta)sin(gamma)cos(alpha) + sin(beta)cos(gamma)sin(alpha)], [cos(beta)cos(gamma)sin(alpha) + sin(beta)sin(gamma)cos(alpha)]]">
+</object>
+
+<div class="example">
+The above quaternion can be represented in JavaScript as follows provided the passed parameters are defined, are {{DeviceOrientationEvent/absolute}} values and those parameters are not null.
+<pre class="lang-json">
```
Comment (reillyeon) [inline index.bs]: s/lang-json/lang-js/

**Comment by anssiko** [comment]
@reillyeon thanks for the review. Your comments have been addressed and fixes squashed to keep the conversion in a single commit https://github.com/w3c/deviceorientation/commit/3ed6c13e0f3d7cb7ec47900e36e1c9fce152c6b4

**Comment by anssiko** [comment]
I'll merge this. Here's hoping nothing breaks badly due to this conversion. I tried to pay attention to the stability of fragment identifiers, but I suspect some autogenerated ids may have slipped by me in the process. We can fix those when we unearth them. The usual suspects linking to fragments are other specs and MDN & friends.

@mkruisselbrink if you identify bugs or issues with this conversion, please feel free to submit PRs with fixes.

**Pull request merged**